examples/arduino-uno/src/bin/uno-usart-embedded-io.rs

```diff
@@ -0,0 +1,29 @@
+/*!
+ * Demonstration of writing to and reading from the serial console.
+ */
+#![no_std]
+#![no_main]
+
+use panic_halt as _;
+
+use embedded_io::{Read, Write};
+
+fn usart_handler(serial: &mut (impl Read + Write)) -> ! {
+    serial.write_all("Hello from Arduino!\r\n".as_bytes()).unwrap();
+
+    loop {
+        let mut rx_buf: [u8; 16] = [0; 16];
+        let len = serial.read(&mut rx_buf).unwrap();
+
+        writeln!(serial, "Got {:?} (which is {} bytes long)", &rx_buf[..len], len).unwrap();
```

Comment:
The problem with this code is that `read()` won't every return more than a few bytes.  Immediately after receiving them, the `writeln!()` will block for a long time to spell out the response.  During this time, we probably receive more data but can't handle it as we're still busy and the hardware doesn't have a (large enough) buffer.

I'm trying to think what a better example could look like.  Maybe waiting for a newline?  Or waiting for a large enough time of silence?  I'd choose whatever variant is easier to implement, so probably the newline...